src/operator/roi_pooling.cc

```diff
@@ -74,7 +76,13 @@ inline void ROIPoolForward(const Tensor<cpu, 4, Dtype> &out,
 
     const Dtype* batch_data = bottom_data + data_size * roi_batch_ind;
 
+    #pragma omp parallel for firstprivate(batch_data, top_data, argmax_data)
     for (int c = 0; c < channels_; ++c) {
+      // Increment all data pointers
+      batch_data += c * data.size(2) * data.size(3);
```

Comment:
please cache the xx.size(2) * you.size(3) result outside of the loops since they are invariant, so as to perform fewer multiplies.

Reply:
if you do that, you can just keep adding that number each pass of c and then you don’t need to do a multiple for these at all

Reply:
multiply*